packages/web/src/components/import-export-srcbook-modal.tsx

```diff
@@ -65,12 +181,94 @@ export function ImportSrcbookModal({
             </p>
           </DialogDescription>
         </DialogHeader>
-        <FilePicker
-          dirname={fsResult.dirname}
-          entries={fsResult.entries}
-          cta="Open"
-          onChange={onChange}
-        />
+
+        <Tabs
+          value={activeTab}
+          onValueChange={(tab) => setActiveTab(tab as 'file' | 'url' | 'clipboard')}
+        >
+          <div className="border-b mb-4">
+            <TabsList>
+              <TabsTrigger value="file">
+                <div className="flex items-center gap-2">
+                  <FilesIcon size={18} />
+                  From Filesystem
+                </div>
+              </TabsTrigger>
+              <TabsTrigger value="url">
+                <div className="flex items-center gap-2">
+                  <GlobeIcon size={18} />
+                  From URL
+                </div>
+              </TabsTrigger>
+              <TabsTrigger value="clipboard">
+                <div className="flex items-center gap-2">
+                  <ClipboardIcon size={18} />
+                  From Clipboard
+                </div>
+              </TabsTrigger>
+            </TabsList>
+          </div>
+
+          <TabsContent className="mt-0" value="file">
+            <SrcMdUploadDropZone onDrop={onCreateSrcbookFromFilesystem} />
+          </TabsContent>
+          <TabsContent className="mt-0" value="url">
+            <div className="flex gap-2 w-full">
+              <Input
+                ref={urlInputRef}
+                value={url}
+                onChange={(event) => setUrl(event.target.value)}
+                placeholder="eg: https://example.com/my-fancy-srcbook.src.md"
+                onKeyDown={(e) => {
+                  if (e.key === 'Enter') {
+                    onCreateSrcbookFromUrl(url);
+                  }
+                }}
+              />
+              <Button
+                type="button"
+                onClick={() => onCreateSrcbookFromUrl(url)}
+                disabled={url.length === 0 || loading}
+              >
+                {loading ? (
+                  <div className="flex items-center gap-2">
+                    <Loader2Icon size={18} className="animate-spin" />
+                    Loading...
+                  </div>
+                ) : (
+                  'Create'
+                )}
+              </Button>
+            </div>
+          </TabsContent>
+          <TabsContent className="mt-0" value="clipboard">
+            <div className="flex flex-col gap-4">
+              <Textarea
+                ref={clipboardTextareaRef}
+                value={clipboard}
+                onChange={(event) => setClipboard(event.target.value)}
+                placeholder="Paste clipboard here"
+                className="h-[112px] resize-none"
```

Comment:
Ok, sounds good!

(as a counterargument to consider, the reason I didn't make this one resizable was because it seemed like it would be rare for a user to want to scroll through the raw text, and would likely only look at a preview and copy it to the clipboard)

Reply:
Yeah it's not a huge thing, most people probably won't. Maybe since I'm a developer of this tool and want to confirm things look right, I'm personally more likely to glance through it